### PR TITLE
For #22310 - Change tab holder's checkbox state when pressing "Select…

### PR DESCRIPTION
**Pull request by Robert-Aron293**
Change tab holder's checkbox state when pressing "Select All"/"Deselect All".

### Pressed "Select All"
Screenshot_20211114-163817_Firefox Preview
### Pressed "Deselect All"
Screenshot_20211114-163824_Firefox Preview

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt [hunk 1]
```diff
@@ -45,9 +45,9 @@ class CollectionCreationTabListAdapter(
                 is CheckChanged -> {
                     val checkChanged = payloads[0] as CheckChanged
                     if (checkChanged.shouldBeChecked) {
```
Comment (Alexandru2909) [inline app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt:47]: Hi, this is just a drive-by comment: 
If you decided to drop true/false and use checkChanged.shouldBeChecked to call holder::changeCheckBoxState, why not also change the if condition?
I was thinking of something like checkChanged.souldBeChecked || checkChanged.shouldBeUnchecked, it is also easier to read.

Comment (Robert-Aron293) [reply app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt]: You are right. Thanks

File: app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt [hunk 2]
```diff
@@ -44,10 +44,8 @@ class CollectionCreationTabListAdapter(
             when (payloads[0]) {
                 is CheckChanged -> {
                     val checkChanged = payloads[0] as CheckChanged
-                    if (checkChanged.shouldBeChecked) {
-                        binding.tabSelectedCheckbox.isChecked = true
-                    } else if (checkChanged.shouldBeUnchecked) {
-                        binding.tabSelectedCheckbox.isChecked = false
+                    if (checkChanged.shouldBeChecked || checkChanged.shouldBeUnchecked) {
+                        holder.changeCheckBoxState(checkChanged.shouldBeChecked)
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt:48]: This check assumes that whenever `shouldBeUnchecked` is true `shouldBeChecked` is false.
Which sounds right as per the naming but the current if/else does not make this assumption and makes me wonder why we had this abstraction in the first place and whether we can just drop the `shouldBeUnchecked` property or it's an important edgecase we need to still acount for.

Comment (Alexandru2909) [reply app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt:48]: @Robert-Aron293 to minimize the risk of regressions let's keep the previous if/else approach for now and I'll investigate how `CheckChanged` is used

**Review by Alexandru2909 — APPROVED** [review]

**Comment by Alexandru2909** [comment]
Changes will be landed in #25315

**Comment by Alexandru2909** [comment]
Closed, as changes were landed in #25315.
